tutorials/deutsch_jozsa/deutsch_jozsa.ipynb

```diff
@@ -0,0 +1,423 @@
+{
```

Comment:
<p>First, I would change the example to be:</p><p> </p><p>qfunc test(res:qbit, x:qnum ){</p><p>&nbsp;res = x&gt; (2**(x.size-1))-1;</p><p>}</p><p> </p><p> </p><p>Then, I would build this paragraph from bottom up a bit more. Something like:</p><p> </p><p>The first step in the algorithm is to encode the function that should be evaluated if it is balanced or constant into the quantum algorithm. As a specific example, a balanced function is demonstrated here. The function is evaluated to be 1 if the quantum number x is greater than the median value of all possible inputs, otherwise it is evaluated to be 0. Upon evaluation, the qubit variable res is initailized with the function output value.</p><p> </p><p> </p><p>I would also change the terminology to fit the phase kickback tutorial, and I would refer to there as what's happening here is a phase kickback, so the first steps can be reviewed more quickly</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Classiq/classiq-library/pull/179/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Reply:
<p>Will add the build paragraph. But I strongly think that the example predicate I've given (res = x&gt;7) will be nicer for a beginner to understand the concept in a simple manner. What do you think?</p>